src/game/enemy/components.rs

```diff
@@ -13,7 +13,20 @@ impl Default for Enemy {
     fn default() -> Self {
         Enemy {
             health: 100,
-            direction: Vec2::default()
+            direction: Vec2::default(),
+        }
+    }
+}
+
+#[derive(Component)]
+pub struct Firing {
+    pub timer: Timer,
+}
+
+impl Default for Firing {
+    fn default() -> Self {
+        Firing {
+            timer: Timer::from_seconds(3.0, TimerMode::Once),
```

Comment:
This is how we determine how long the shooting animation goes for (in combination of the AnimationTimer). I tried lowering it to loop once, but it is pretty quick and unnoticeable. I think we can tweak this as we go